### ISSUE TITLE
[Bug]: Integration uses deprecated HASS temperature utility

### ISSUE THREAD

**Issue by teskanoo** [custom-components/ble_monitor#1031]
### What happened?

I'm sure you're aware of this but couldn't see it posted as a bug report so adding it now

This appeared in my Homeassistant error log after upgrading to 2022.10

Had they managed to get the dates in the right order (they are clearly reversed) we may all have known about this back in April :)

`
Detected integration that uses temperature utility. This is deprecated since 2022.10 and will stop working in Home Assistant 2022.4, it should be updated to use unit_conversion.TemperatureConverter instead. Please report issue to the custom integration author for ble_monitor using this method at custom_components/ble_monitor/sensor.py, line 497: self._state = convert_temp(
`

### Sensor type

Any temperature sensor

### Relevant log output

```shell
Detected integration that uses temperature utility. This is deprecated since 2022.10 and will stop working in Home Assistant 2022.4, it should be updated to use unit_conversion.TemperatureConverter instead. Please report issue to the custom integration author for ble_monitor using this method at custom_components/ble_monitor/sensor.py, line 497: self._state = convert_temp(
```

**Comment by Ernst79** [comment]
Thanks, I wasn't aware that this was affecting BLE monitor, so thanks for the issue report. Will look into it. 

I see a bug in the error message, is will stop working in 2022.4 😄

**Comment by Ernst79** [comment]
Fixed in 10.5.0
